src/gui/TextEditor.cpp

```diff
@@ -183,6 +183,13 @@ void TextEditor::iMPreeditChangedCallback(GtkIMContext* context, TextEditor* te)
      */
     gtk_im_context_get_preedit_string(context, &str, &attrs, &cursor_pos);
 
+    if (attrs == nullptr) {
+        attrs = pango_attr_list_new();
+    }
+    te->preeditAttrList = attrs;
+    pango_attr_list_unref(attrs);
```

Comment:
This `pango_attr_list_unref` might be causing the new crash you got. AFAICT ownership is transferred to `te->preeditattrList`, so this unref will drop the reference count to 0 and then lead to double free! I

Reply:
Bingo! Dropping the ``pango_attr_list_unref`` makes ``uim`` work properly.

Reply:
Instead ``pango_attr_list_unref(te->preeditAttrList);`` is needed before ``te->preeditAttrList = attrs;``, isn't it?

Reply:
I think the `pango_attr_list_unref` line should be removed entirely. If you call it twice, you get a double-free segfault, which suggests that calling the unref here is going to free attrs.

Reply:
All right. I will remove the line.